v1/src/test/java/com/google/cloud/teleport/templates/PubsubToAvroLT.java

```diff
@@ -0,0 +1,172 @@
+import static com.google.cloud.teleport.it.common.matchers.TemplateAsserts.assertThatResult;
+import static com.google.cloud.teleport.it.gcp.artifacts.utils.ArtifactUtils.createStorageClient;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.storage.Storage;
+import com.google.cloud.teleport.it.common.PipelineLauncher;
+import com.google.cloud.teleport.it.common.PipelineOperator;
+import com.google.cloud.teleport.it.common.TestProperties;
+import com.google.cloud.teleport.it.common.utils.ResourceManagerUtils;
+import com.google.cloud.teleport.it.gcp.TemplateLoadTestBase;
+import com.google.cloud.teleport.it.gcp.artifacts.ArtifactClient;
+import com.google.cloud.teleport.it.gcp.artifacts.GcsArtifactClient;
+import com.google.cloud.teleport.it.gcp.artifacts.utils.ArtifactUtils;
+import com.google.cloud.teleport.it.gcp.datagenerator.DataGenerator;
+import com.google.cloud.teleport.it.gcp.pubsub.DefaultPubsubResourceManager;
+import com.google.cloud.teleport.it.gcp.pubsub.PubsubResourceManager;
+import com.google.cloud.teleport.metadata.TemplateLoadTest;
+import com.google.common.base.MoreObjects;
+import com.google.pubsub.v1.TopicName;
+import java.io.IOException;
+import java.text.ParseException;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Performance tests for {@link PubsubToAvro} PubSub to GCS Avro template. */
+@Category(TemplateLoadTest.class)
+@TemplateLoadTest(PubsubToAvro.class)
+@RunWith(JUnit4.class)
+public class PubsubToAvroLT extends TemplateLoadTestBase {
+
+  private static final String SPEC_PATH =
+      MoreObjects.firstNonNull(
+          TestProperties.specPath(),
+          "gs://dataflow-templates-test-bucket/templates/Cloud_PubSub_to_Avro");
+  private static final String ARTIFACT_BUCKET = TestProperties.artifactBucket();
+  private static final String TEST_ROOT_DIR = PubsubToAvroLT.class.getSimpleName().toLowerCase();
+  private static final String AVRO_TEMP_DIR = "avro_tmp";
+  private static final String INPUT_PCOLLECTION = "Read PubSub Events/MapElements/Map.out0";
+  private static final String OUTPUT_PCOLLECTION = "5m Window/Window.Assign.out0";
+  private static final String AVRO_OUTPUT_FILENAME_PREFIX = "topic-output-";
+  // 35,000,000 messages of the given schema make up approximately 10GB
+  private static final long NUM_MESSAGES_FOR_10_GB_TEST = 35000000L;
+  private static final long NUM_MESSAGES_FOR_10_MB_TEST = 35000L;
+  private static final int WAIT_FOR_WORKERS_TIME_SEC = 300;
+  private static final long TIMEOUT_MIN_FOR_10_MB_TEST = 20;
+  private static final long TIMEOUT_MIN_FOR_10_GB_TEST = 60;
```

Comment:
`TIMEOUT_MIN_FOR_10_GB_TEST` - why "MIN"? Let's leave just 'TIMEOUT_FOR_10_GB_TEST'

Reply:
I mean in minutes